File: fib.m
//Max Rogers
//Objective-C Recusive Fibonacci C-Style
//to build:
//clang fib.m -lobjc -framework Foundation

#import <Foundation/Foundation.h>

int fib(int n) {
  if (n <= 0) { return 0; }
  else if (n <= 2) { return 1;}
  else { return fib(n-2) + fib(n-1); }
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
      int n = 30;

      if (argc > 1) {
        n = atoi(argv[1]);
      }

      double startTime = (double)clock()/CLOCKS_PER_SEC;
      int x = fib(n);
      double endTime = (double)clock()/CLOCKS_PER_SEC;
      double timeElapsed = endTime - startTime;
      printf("ObjC\tFibonacci:\t%d\tExecTime: %.2Es\tn:%d",x,timeElapsed,n);
    }
    return 0;
}

File: fib.swift
//
//  fib.swift
//  SwiftFibBenchmark
//
//  Created by Max Rogers on 11/17/15.
//  Copyright © 2015 Max Rogers. All rights reserved.
//

import Foundation

func fib(n: Int) -> Int {
    if n <= 0 {
        return 0
    } else if n <= 2 {
        return 1
    } else {
        return fib(n:n-2) + fib(n:n-1)
    }
}

var n = 30
if Process.arguments.count > 1 {
    n = Int(Process.arguments[1])!
}
let startTime = CFAbsoluteTimeGetCurrent()
let x = fib(n:n)
let timeElapsed = CFAbsoluteTimeGetCurrent() - startTime
let numberFormatter = NumberFormatter()
numberFormatter.numberStyle = NumberFormatter.Style.scientific
numberFormatter.maximumSignificantDigits = 3
let timeString = numberFormatter.string(from:timeElapsed)
print("SWIFT\tFibonacci:\t\(x)\tExecTime: \(timeString!)s\tn:\(n)")
